### PR TITLE
GUI-2603, update button scss mixin, tighten submit strictures.

### PR DESCRIPTION
**Pull request by coyote240**
https://eucalyptus.atlassian.net/browse/GUI-2603

### CODE REVIEW

File: eucaconsole/static/js/pages/custom_filters.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ angular.module('CustomFilters', [])
 })
 .filter('ellipsis', function () {
     return function (line, num) {
-        if( line === null || line.length === 0 ){
+        if(!line) {
```
Comment (coyote240) [inline eucaconsole/static/js/pages/custom_filters.js]: Missing that one, critical check, is it undefined?  All three are captures in truthiness.

**Comment by coyote240** [comment]
retest this please

**Comment by kamalgill** [comment]
Looks good to me, passing to @dkavanagh for another stamp of approval...

File: eucaconsole/static/js/widgets/tag-editor/tag-editor.js [hunk 2]
```diff
@@ -86,7 +91,7 @@ angular.module('TagEditorModule', ['EucaConsoleUtils'])
         };
     })
     .directive('tagValue', function () {
-        var validPattern = /^(?!aws:).{0,256}$/;
+        var validPattern = /^(?!aws:).{1,256}$/;
```
Comment (dkavanagh) [inline eucaconsole/static/js/widgets/tag-editor/tag-editor.js]: Are we handling the empty tag value here? In that case, 0 lower end would be OK.

Comment (kamalgill) [inline eucaconsole/static/js/widgets/tag-editor/tag-editor.js]: We don't support tags with empty values yet.  See comments in https://eucalyptus.atlassian.net/browse/GUI-2603

File: eucaconsole/templates/tag-editor/tag-editor.pt [hunk 3]
```diff
@@ -39,18 +39,17 @@
                         Tag names must be unique
                     </span>
                 </div>
-                <div id="autoscale-tag-value-input-div" class="large-8 medium-7 small-5 columns">
+                <div id="autoscale-tag-value-input-div" class="large-6 medium-6 small-4 columns">
```
Comment (dkavanagh) [inline eucaconsole/templates/tag-editor/tag-editor.pt]: I'll point out a feature of this grid system that if in all views, the number of columns is the same, you only need to specify the small. In this case, you could get away with "medium-6 small-4" and get the same results.

**Comment by dkavanagh** [comment]
retest this please

**Comment by aliceh** [comment]
Retest this please

**Comment by aliceh** [comment]
Retest this please

**Comment by aliceh** [comment]
@kamalgill  @coyote240 In this branch, when desired capacity, min or max of autoscaling group is changed the "Save changes" button does not become active. That is the reason for the consitent failures in  autoscaling sequence.

File: eucaconsole/static/js/widgets/tag-editor/tag-editor.js [hunk 4]
```diff
@@ -75,7 +80,7 @@ angular.module('TagEditorModule', ['EucaConsoleUtils'])
         };
     }])
     .directive('tagName', function () {
-        var validPattern = /^(?!aws:)(?!euca:).{0,128}$/;
+        var validPattern = /^(?!aws:)(?!euca:).{1,128}$/;
```
Comment (dkavanagh) [inline eucaconsole/static/js/widgets/tag-editor/tag-editor.js]: Requiring values in this field (and the value field) causes the form $invalid flag to be true. That disables the SaveChanges button when it should be enabled.

**Comment by aliceh** [comment]
Retest this please

**Comment by dkavanagh** [comment]
@aliceh nothing was fixed. Not worth retesting till Adam gets a chance to look at this tomorrow.

**Comment by dkavanagh** [comment]
Changes make sense. Allowing 0 size for field level validation, but disallowing add tag if values aren't populated.

**Pull request merged**